br/pkg/lightning/backend/local/local.go

```diff
@@ -1541,19 +1664,54 @@ func (local *local) writeAndIngestByRanges(ctx context.Context, engine *Engine,
 	var allErr error
 	var wg sync.WaitGroup
 	metErr := atomic.NewBool(false)
+	rangeAndRegions := make([]RangeAndRegion, 0)
+	regionProcessingMap := sync.Map{}
 
 	for _, r := range ranges {
 		startKey := r.start
 		endKey := r.end
+		regions, err := local.scanRegion(ctx, engine, startKey, endKey, regionSplitSize, regionSplitKeys)
+		if err != nil {
+			log.FromContext(ctx).Warn("scaen regino failed", zap.Error(err))
+			return err
+		}
+		rr := RangeAndRegion{
+			r:           r,
+			isScheduled: false,
+			regions:     make(map[uint64]int),
+		}
+		for _, region := range regions {
+			for _, peer := range region.Region.Peers {
+				if val, ok := rr.regions[peer.StoreId]; ok {
+					rr.regions[peer.StoreId] = val + 1
+				} else {
+					rr.regions[peer.StoreId] = 1
+				}
+			}
+		}
+		rangeAndRegions = append(rangeAndRegions, rr)
+	}
+
+	finished := 0
+outerLoop:
+	for finished < len(rangeAndRegions) {
+		rr := pickRanges(rangeAndRegions, &regionProcessingMap, finished)
+		startKey := rr.r.start
+		endKey := rr.r.end
+		processingRegions := rr.regions
```

Comment:
passing `processingRegions` to `writeAndIngestByRange` in line 1722, to skip the inner scan region API in line 1470?